torcharrow/velox_rt/numerical_column_cpu.py

```diff
@@ -595,23 +595,13 @@ def log(self):
     @trace
     @expression
     def fill_null(self, fill_value: Union[dt.ScalarTypes, Dict]):
-        self._prototype_support_warning("fill_null")
-
         if not isinstance(fill_value, Column._scalar_types):
             raise TypeError(f"fill_null with {type(fill_value)} is not supported")
         if not self.is_nullable:
             return self
-        else:
-            col = velox.Column(get_velox_type(self.dtype))
-            for i in range(len(self)):
-                if self._getmask(i):
-                    if isinstance(fill_value, Dict):
-                        raise NotImplementedError()
-                    else:
-                        col.append(fill_value)
-                else:
-                    col.append(self._getdata(i))
-            return ColumnCpuMixin._from_velox(self.device, self.dtype, col, True)
+
+        fill_value_casted = dt.np_typeof_dtype(self.dtype)(fill_value)
```

Comment:
add a comment like 
```
# functional doesn't support type promotion yet. Cast to column dtype manually (represented by numpy literal) 
```
? 

Also let's add some basic tests (e.g. `fill_null` test for int32 and float64 column.